src/proof/theory_proof.cpp

```diff
@@ -1264,4 +1263,387 @@ void TheoryProof::printRewriteProof(std::ostream& os, const Node &n1, const Node
+      i += count;
+      if (targetAppearsAfter)
+      {
+        ++i;
+      }
+    }
+
+    // Else, just copy the child proof as is
+    else
+    {
+      subTrans->d_children.push_back(pf.d_children[i]);
+      ++i;
+    }
+  }
+
+  bool disequalityFound = (*neg >= 0);
+  if (!disequalityFound)
+  {
+    Debug("pf::" + theoryName)
+        << "A disequality was NOT found. UNSAT due to merged constants"
+        << std::endl;
+    Debug("pf::" + theoryName) << "Proof for: " << pf.d_node << std::endl;
+    Assert(pf.d_node.getKind() == kind::EQUAL);
+    Assert(pf.d_node.getNumChildren() == 2);
+    Assert(pf.d_node[0].isConst() && pf.d_node[1].isConst());
+  }
+}
+
+void TheoryProof::identicalEqualitiesPrinterHelper(theory::TheoryId theoryId,
+                                            bool evenLengthSequence,
+                                            bool sequenceOver,
+                                            int i,
+                                            const theory::eq::EqProof& pf,
+                                            const ProofLetMap& map,
+                                            const Node& n2,
+                                            const std::string ss1String,
+                                            std::stringstream* ss,
+                                            Node& n1,
+                                            Node& nodeAfterEqualitySequence)
+{
+  bool ufProof = (theoryId == theory::THEORY_UF);
+  std::string theoryName;
+  if (theoryId == theory::THEORY_UF)
+  {
+    theoryName = "uf";
+  }
+  else
+  {
+    theoryName = "array";
+  }
+  if (evenLengthSequence)
+  {
+    // If the length is even, we need to apply transitivity for the "correct"
+    // hand of the equality.
+
+    Debug("pf::" + theoryName) << "Equality sequence of even length"
+                               << std::endl;
+    Debug("pf::" + theoryName) << "n1 is: " << n1 << std::endl;
+    Debug("pf::" + theoryName) << "n2 is: " << n2 << std::endl;
+    Debug("pf::" + theoryName) << "pf-d_node is: " << pf.d_node << std::endl;
+    Debug("pf::" + theoryName) << "Next node is: " << nodeAfterEqualitySequence
+                               << std::endl;
+
+    (*ss) << "(trans _ _ _ _ ";
+
+    // If the sequence is at the very end of the transitivity proof, use
+    // pf.d_node to guide us.
+    if (!sequenceOver)
+    {
+      if (match(n1[0], pf.d_node[0], theoryId))
+      {
+        n1 = n1[0].eqNode(n1[0]);
+        (*ss) << ss1String << " (symm _ _ _ " << ss1String << ")";
+      }
+      else if (match(n1[1], pf.d_node[1], theoryId))
+      {
+        n1 = n1[1].eqNode(n1[1]);
+        (*ss) << " (symm _ _ _ " << ss1String << ")" << ss1String;
+      }
+      else
+      {
+        Debug("pf::" + theoryName) << "Error: identical equalities over, but "
+                                      "hands don't match what we're proving."
+                                   << std::endl;
+        Assert(false);
```

Comment:
We should change those to [`CVC4_CHECK`](https://github.com/CVC4/CVC4/blob/master/src/base/cvc4_check.h#L131) (not necessarily in this PR).